balloon/src/error.rs

```diff
@@ -0,0 +1,49 @@
+//! Error type
+
+use core::fmt;
+
+#[cfg(feature = "password-hash")]
+use password_hash::errors::InvalidValue;
+
+/// Result with balloon's [`Error`] type.
+pub type Result<T> = core::result::Result<T, Error>;
+
+/// Error type.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Error {
+    /// Memory cost is too small.
+    MemoryTooLittle,
+    /// Not enough threads.
+    ThreadsTooFew,
+    /// Too many threads.
+    ThreadsTooMany,
+    /// Time cost is too small.
+    TimeTooSmall,
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self {
+            Error::MemoryTooLittle => "memory cost is too small",
+            Error::ThreadsTooFew => "not enough threads",
+            Error::ThreadsTooMany => "too many threads",
+            Error::TimeTooSmall => "time cost is too small",
```

Comment:
I could add more descriptive messages here. `MemoryTooLittle` could mention how much memory is actually needed. `ThreadsTooMany` could mention that you need the `parallel` crate feature to use threads. I don't know how to translate any of this to `password_hash::Error` though.

Reply:
I added how much memory is needed in the documentation.